File: src/main/java/fr/marstech/mtlinkspray/controller/api/DashboardApiController.java
package fr.marstech.mtlinkspray.controller.api;

import fr.marstech.mtlinkspray.dto.DashboardDto;
import fr.marstech.mtlinkspray.service.DashboardService;
import org.springframework.web.bind.annotation.*;

@RestController
public class DashboardApiController {

    private final DashboardService dashboardService;

    public DashboardApiController(DashboardService dashboardService) {
        this.dashboardService = dashboardService;
    }

    @GetMapping("/api/dashboard/{id}")
    public DashboardDto getDashboard(@PathVariable(name = "id") String id) {
        return dashboardService.getDashboard(id);
    }

    @PostMapping("/api/dashboard")
    public DashboardDto createDashboard(DashboardDto dashboardDto) {
        return dashboardService.createDashboard(dashboardDto);
    }

    @PutMapping("/api/dashboard/{id}")
    public DashboardDto updateDashboard(@PathVariable(name = "id") String id, @RequestBody DashboardDto dashboardDto) {
        return dashboardService.updateDashboard(id, dashboardDto);
    }
}

File: src/main/kotlin/fr/marstech/mtlinkspray/controller/api/DashboardApiController.kt
package fr.marstech.mtlinkspray.controller.api

import fr.marstech.mtlinkspray.dto.DashboardDto
import fr.marstech.mtlinkspray.service.DashboardService
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@Validated
@RestController
@RequestMapping("/api/dashboard")
class DashboardApiController(
    private val dashboardService: DashboardService
) {

    @GetMapping("/{id}")
    @Validated
    fun getDashboard(
        @PathVariable(name = "id") @NotBlank(message = "ID cannot be blank") id: String
    ): DashboardDto = dashboardService.getDashboard(id)

    @PostMapping
    fun createDashboard(
        @RequestBody @Valid dashboardDto: DashboardDto
    ): DashboardDto = dashboardService.createDashboard(dashboardDto)

    @PutMapping("/{id}")
    @Validated
    fun updateDashboard(
        @PathVariable(name = "id") @NotBlank(message = "ID cannot be blank") id: String,
        @RequestBody @Valid dashboardDto: DashboardDto
    ): DashboardDto = dashboardService.updateDashboard(id, dashboardDto)
}
